// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

syntax = "proto2";

message t_ul_stats_item {
   required uint32 count = 1;
   required uint32 total = 2;
   required uint32 min = 3;
   required uint32 max = 4;
}

message t_ul_traffic_stats_item {
   required uint32 count = 1;
   required uint32 total = 2;
}

message t_ul_stats {
   required uint32 du_ue_index = 1;
   required uint32 is_srb = 2;
   required uint32 rb_id = 3;
   required t_ul_stats_item pdu_window = 4;
   required t_ul_traffic_stats_item pdu_bytes = 5;
   required t_ul_traffic_stats_item sdu_delivered_bytes = 6;
}

message rlc_ul_stats {
   required uint64 timestamp = 1;
   repeated t_ul_stats stats = 2;
}




